keras/src/ops/function.py

```diff
@@ -81,6 +81,32 @@ def __init__(self, inputs, outputs, name=None):
         self._nodes_by_depth = nodes_by_depth
         self._operations = operations
         self._operations_by_depth = operations_by_depth
+        if backend() == "openvino":
```

Comment:
Hi @fchollet, OpenVINO does not support eager mode for inference. We should build OV graph preliminary and run inference for the whole graph. So I decided to construct a graph in `init` and use it for `call`. Please propose how to avoid backend-specific code here due to this OV specific.

Reply:
Just disable what doesn't work, with a clear error message, at the level of the openvino backend. If openvino is only usable via `evaluate` and `predict`, that's ok.

Though tbh since this backend is inference only and doesn't have eager support, it sounds like maybe it should be a backend, but instead should be an export format in `model.export()`.

Reply:
@fchollet, please clarify. Your point is to exclude this backend-specific code and configure the backend via `_...` variables. This configuration should tell Keras 3 that OpenVINO only works in non-eager mode. Am I correct? I just want ops helpers to work only with symbolic tensors `ov::Output` or `ov::Node` instances (for graph construction). And I want this graph to be constructed only one for one device and input shape specification. Is it possible to do in Keras 3?

Or your point only to implement `export` method? It will allow only to infer using OpenVINO API after model loading and not Keras 3. Please correct me.

Reply:
So there are two entirely separate options:

1. Make an openvino backend. If we do this it should have roughly the same scope of functionality as the numpy backend. It should "fit" the backend format. I am not sure this is feasible if openvino has no support for eager execution.
2. Make `model.export()`/`ExportArchive` able to save a model in the openvino format (the model would be coming from either Keras + JAX, or Keras + torch, of Keras + TF). That was folks can train their models in another backend, then export it to openvino.

Reply:
Hi @fchollet, thanks a lot for explanation!
Our original plan has been to have OpenVINO as backend. It helps users do not override keras code and infer via OpenVINO. Different code snippets for Keras models (NLP, CV models) would be inferred by OpenVINO. `Export` option makes user to write OpenVINO code after export for these models that is less convenient, unfortunately.

I am thinkng about implementing functionality (for graph construction and compilation) in OpenVINO trainer class but I do not see how to access to Keras model nodes. I see inference is triggered via callback mechanism. I have two questions:
1. Is any option to access Keras model nodes from Trainer class?
2. Is any option to configure inference callback in Trainer class to set my custom one implemented in OpenVINO Trainer class?
3. Do you have any plan to extend Keras developer API for adding inference frameworks that do not support eager mode? I think for optimal inference non-eager mode can be more optimal than eager mode.

Thanks a lot for your help,
Roman

Reply:
How do you create a compiled graph in OpenVINO?

Reply:
Here is an example of how `compiled_model` is created:

```
import openvino.runtime.opset9 as ov
from openvino.runtime import Model, Core

# create a model with TopK
x = ov.parameter([1], name="x", dtype=np.int32)
y = ov.parameter([1], name="y", dtype=np.int32)
divide = ov.divide(x, y)
ov_model = Model([divide], [x, y], "model")

# compile the model for CPU device
core = ov.Core()
compiled_model = core.compile_model(ov_model, 'CPU')
```

At the first step, we build `ov::Model` ov_model object and compile then.
 @fchollet, may be I can set up a meeting with you on Discord (@rkazants on Discord) to discuss further steps and potential solution?

Really appreciate your time and help,
Roman

Reply:
And then you call the `compiled_model` on numpy data?

That sort of logic would belong in the `Trainer` class. You can try prototyping an OpenVINO Trainer that works like this.

Reply:
> And then you call the `compiled_model` on numpy data?

Exactly, it calls `compiled_model` on numpy data for prediction

Reply:
If we implement a backend like this, it won't be usable eagerly, but only through `Model.evaluate()` and `Model.predict()`. That's quite limiting. It's still feasible though. We won't be able to test it on CI, since most of our tests need eager mode, instead we'd need a new set of integration tests specifically for this backend.

Reply:
For inference such functionality is sufficient) Let me try to do this.

Best regards,
Roman